packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -359,14 +360,15 @@ async function extractCustomPackageData(pkg: Package, fetchResult: FetchResult)
   };
 }
 
-async function writeInstallState(project: Project, locatorMap: NodeModulesLocatorMap, binSymlinks: BinSymlinkMap) {
+async function writeInstallState(project: Project, locatorMap: NodeModulesLocatorMap, binSymlinks: BinSymlinkMap, hardlinks: boolean) {
```

Comment:
I'd recommend to make it an object parameter, to avoid the [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap) (especially since you use `useHardlinks` this way further below).